xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import io.grpc.Attributes;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.NameResolver;
+import io.grpc.Status;
+import io.grpc.SynchronizationContext;
+import io.grpc.internal.GrpcAttributes;
+import io.grpc.internal.JsonParser;
+import io.grpc.internal.SharedResourceHolder;
+import io.grpc.internal.SharedResourceHolder.Resource;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+/**
+ * A {@link NameResolver} for resolving gRPC target names with "xds" scheme.
+ *
+ * <p>The implementation is for load balancing alpha release only. No actual VHDS is involved. It
+ * always returns a hard-coded service config that selects the xds_experimental LB policy with
+ * round-robin as the child policy.
+ *
+ * @see XdsNameResolverProvider
+ */
+public final class XdsNameResolver extends NameResolver {
+
+  private static final String SERVICE_CONFIG_HARDCODED = "{"
+          + "\"loadBalancingConfig\": ["
+          + "{\"xds_experimental\" : {"
+          + "\"childPolicy\" : [{\"round_robin\" : {}}]"
+          + "}}"
+          + "]}";
+
+  private final String authority;
+  private final Resource<Executor> executorResource;
+  private final SynchronizationContext syncContext;
+
+  private boolean resolving;
+  private boolean shutdown;
+  private Executor executor;
+  private Listener2 listener;
+
+  XdsNameResolver(String name, Args args, Resource<Executor> executorResource) {
+    URI nameUri = URI.create("//" + checkNotNull(name, "name"));
+    Preconditions.checkArgument(nameUri.getHost() != null, "Invalid hostname: %s", name);
+    authority =
+        Preconditions.checkNotNull(
+            nameUri.getAuthority(), "nameUri (%s) doesn't have an authority", nameUri);
+    this.executorResource = Preconditions.checkNotNull(executorResource, "executorResource");
```

Comment:
This executor was for blocking operations in DNS. I don't think we have need for it here.

Reply:
Sure. I was thinking about having the same behavior as DNS resolver that the listener receives resolution result from a separate thread. But, if the alpha resolver aims to be so "fake", everything can just run and return straightly.